src/Hosting/Hosting/src/WebHostExtensions.cs

```diff
@@ -135,18 +135,18 @@ private static void AttachCtrlcSigtermShutdown(CancellationTokenSource cts, Manu
         {
             void Shutdown()
             {
-                if (!cts.IsCancellationRequested)
+                try
                 {
-                    if (!string.IsNullOrEmpty(shutdownMessage))
-                    {
-                        Console.WriteLine(shutdownMessage);
-                    }
-                    try
+                    if (!cts.IsCancellationRequested)
                     {
+                        if (!string.IsNullOrEmpty(shutdownMessage))
+                        {
+                            Console.WriteLine(shutdownMessage);
+                        }
                         cts.Cancel();
                     }
-                    catch (ObjectDisposedException) { }
                 }
+                catch (Exception) { }
```

Comment:
Needs a comment on why this might throw.

Reply:
Especially since someone might come along and be like "Hrm, we should at least log the exception to the console" 💥